### ISSUE TITLE
Toggle on and off via AppleScript

### ISSUE THREAD

**Issue by drewbarontini** [newmarcel/KeepingYouAwake#43]
Hello!

Previously, with Caffeine, I was able to use the following AppleScript to toggle it on and off via the keyboard:

```applescript
tell application "System Events"
	if exists (process "Caffeine") then
		tell application "Caffeine"
			if active is false then
				turn on
			else
				turn off
			end if
		end tell
	else
		tell application "Caffeine" to activate
	end if
end tell
```

When trying that with this application (subbing `"Caffeine"` with `"KeepingYouAwake"`), it doesn't work. Using the Script Editor application, it seems that the `active` variable is not present. As far as I can tell, there aren't any available actions when using the Script Editor Dictionary, but I thought I'd confirm here.

**Comment by newmarcel** [comment]
You are right, I should probably integrate some scripting definitions to be a good Cocoa and AppleScript citizen.

I'll do some research if it is easily possible to ship basic AppleScript support in the next release. I'll report any findings here.

Thanks for reminding me :grinning:

**Comment by drewbarontini** [comment]
Awesome! Thank you, sir :+1:

**Comment by hgiesel** [comment]
Would also love to see this feature. It would just need "turnOn", "turnOff" and "getCurrentStatus", to enable toggling. This would be completely enough.

**Comment by newmarcel** [comment]
I'm sorry for the delay, but I became absolutely discouraged from developing this feature when I saw how these AppleScript APIs and scripting definitions should be implemented 😭.

I'll have a look at this year's AppleScript-related WWDC sessions to see if there is anything new in there to make my life easier implementing scripting support.

I'll update this issue when I've gathered new insights into AppleScript. Thanks for your patience!

**Comment by marcphk** [comment]
Just to add - I would very much like this too if it's something that's still on the cards.

**Comment by tsankuanglee** [comment]
Or is there a way to do it in `bash`? Having an API-like service can open up possibilities such as user-defined scripts (therefore a lot of custom features can be offload to user scripts), control from the touch bar, etc.

**Comment by babul** [comment]
My use case was I wanted to reset the countdown to 5 hours via a bash script, which is how I got here.

To workaround not being able to AppleScript it, I just set the default activation period to 5 hours, then close and reopen the app from Bash. Easy peasy!

**Comment by AlecRust** [comment]
Is there still not a simple way to toggle KeepingYouAwake from the command line?

I've been using Lungo only because it [has a CLI](https://github.com/sindresorhus/lungo-cli) for this.

**Comment by newmarcel** [comment]
@AlecRust  Since version [1.3.0](https://github.com/newmarcel/KeepingYouAwake/blob/main/CHANGELOG.md#v13) you can use URL schemes to control KYA from the command line, e.g.

```
open "keepingyouawake:///activate"
open "keepingyouawake:///deactivate"
open "keepingyouawake:///activate?minutes=5"
```
_(please note the quotes when using zsh)_

Is that insufficient for your use case? If that is the case I would even recommend using Apple's [`caffeinate`](https://web.archive.org/web/20140604153141/https://developer.apple.com/library/mac/documentation/Darwin/Reference/ManPages/man8/caffeinate.8.html) command line tool directly.

**Comment by AlecRust** [comment]
Oh that's great, thanks!

**Comment by alexraileanu** [comment]
it wasn't entirely obvious to me but looking thru the source code, i found out that there is also `open "keepingyouawake:///toggle"` to, you guessed it, toggle activation state.

**Comment by RX4NT9UP** [comment]
FYI  on 1.6.5 at least to make this work consistently I had to do
`open “keepingyouawake:///deactivate” && open “keepingyouawake:///activate”`

**Comment by paulmoons** [comment]
I'm not sure why but when I open up Script Editor and enter this command to test it, I get the following error. MacOS 26; KeepingYouAwake v1.6.8. 

What am I doing wrong?

<img width="703" height="736" alt="Image" src="https://github.com/user-attachments/assets/72a5b922-b536-4aab-a684-2d5b8ee06256" />
